plugins/modules/packaging/os/snap.py

```diff
@@ -186,36 +266,92 @@ def is_snap_enabled(self, snap_name):
         return "disabled" not in notes.split(',')
 
     def state_present(self):
+
         self.vars.meta('classic').set(output=True)
         self.vars.meta('channel').set(output=True)
         actionable_snaps = [s for s in self.vars.name if not self.is_snap_installed(s)]
-        if not actionable_snaps:
-            return
-        self.changed = True
-        self.vars.snaps_installed = actionable_snaps
-        if self.module.check_mode:
-            return
-        params = ['state', 'classic', 'channel']  # get base cmd parts
-        has_one_pkg_params = bool(self.vars.classic) or self.vars.channel != 'stable'
-        has_multiple_snaps = len(actionable_snaps) > 1
-        if has_one_pkg_params and has_multiple_snaps:
-            commands = [params + [{'actionable_snaps': [s]}] for s in actionable_snaps]
-        else:
-            commands = [params + [{'actionable_snaps': actionable_snaps}]]
-        self.vars.cmd, rc, out, err = self._run_multiple_commands(commands)
-        if rc == 0:
-            return
 
-        classic_snap_pattern = re.compile(r'^error: This revision of snap "(?P<package_name>\w+)"'
-                                          r' was published using classic confinement')
-        match = classic_snap_pattern.match(err)
-        if match:
-            err_pkg = match.group('package_name')
-            msg = "Couldn't install {name} because it requires classic confinement".format(name=err_pkg)
-        else:
-            msg = "Ooops! Snap installation failed while executing '{cmd}', please examine logs and " \
-                  "error output for more details.".format(cmd=self.vars.cmd)
-        raise ModuleHelperException(msg=msg)
+        if actionable_snaps:
+            self.changed = True
```

Comment:
Maybe move this into a separate function?

Reply:
I do not understand which part should be moved into a separate function. Most of the method is old. My part (self.set_options()) is already in a new function. Could you elaborate?

Reply:
Everything that's inside the `if` block. Having this huge chunk indented makes it a lot harder to follow what this function does (and also the `return`s inside that `if` block seem wrong in some cases when they are in this `if`). (And yes, I know it is old code.)

Reply:
Moved to a separate function.